Sources/Target/Common/ThreadBase.cpp

```diff
@@ -30,5 +30,17 @@ ErrorCode ThreadBase::modifyRegisters(
   action(state);
   return writeCPUState(state);
 }
+
+ErrorCode ThreadBase::beforeResume() {
+  if (!_process->isAlive())
+    return kErrorProcessNotFound;
+
+  BreakpointManager *bpm = _process->hardwareBreakpointManager();
+  if (bpm != nullptr) {
+    bpm->enable();
```

Comment:
Change the API then? :laughing:

I don't think re-enabling all the watchpoints on all the threads every time
a thread is created is a good solution.

On Tuesday, August 9, 2016, Francis Ricci <notifications@github.com> wrote:

> In Sources/Target/Common/ThreadBase.cpp
> <https://github.com/facebook/ds2/pull/243#discussion_r74100354>:
>
> > @@ -30,5 +30,17 @@ ErrorCode ThreadBase::modifyRegisters(
> >    action(state);
> >    return writeCPUState(state);
> >  }
> > +
> > +ErrorCode ThreadBase::beforeResume() {
> > +  if (!_process->isAlive())
> > +    return kErrorProcessNotFound;
> > +
> > +  BreakpointManager *bpm = _process->hardwareBreakpointManager();
> > +  if (bpm != nullptr) {
> > +    bpm->enable();
>
> That API requires pointers to the actual breakpoint Sites, which only the
> superclass has.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/facebook/ds2/pull/243/files/2a73b10db4691d10fd51d07ec46fbded7b36f6af..b32f1128fa8ceb0010cf5f79262d062aca477162#r74100354>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AADHbBnJzSYUxJ7OLPWM1nxAxSUAn82nks5qeLNGgaJpZM4H9FRy>
> .
>


-- 
Stephane Sezer